orthopairs/src/main/java/org/reactome/release/orthopairs/AlternateIdMapper.java

```diff
@@ -0,0 +1,169 @@
+package org.reactome.release.orthopairs;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.*;
+import java.util.*;
+
+public class AlternateIdMapper {
+    private static final Logger logger = LogManager.getLogger();
+    // Since we require some species-specific databases, there is a method for mapping the returned files. This wasn't
+    // abstractble, considering that the files all have different formats.
+    public static Map<String, Set<String>> getAltIdMappingFile(Object speciesKey, String alternateIdFilename) throws IOException {
+
+        File alternateIdFile = new File(alternateIdFilename);
+        BufferedReader br = new BufferedReader(new FileReader(alternateIdFile));
+        Map<String,Set<String>> altIdToEnsemblMap = new HashMap<>();
+        if (speciesKey.equals("mmus")) {
+            altIdToEnsemblMap = mapMouseAlternateIds(br);
+        } else if (speciesKey.equals("rnor")) {
+            altIdToEnsemblMap = mapRatAlternateIds(br);
+        } else if (speciesKey.equals("xtro")) {
+           altIdToEnsemblMap = mapFrogAlternateIds(br);
+        } else if (speciesKey.equals("drer")) {
+            altIdToEnsemblMap = mapZebraFishAlternateIds(br);
+        } else if (speciesKey.equals("scer")) {
+            altIdToEnsemblMap = mapYeastAlternateIds(br);
+        } else {
+            logger.warn(speciesKey + " does not have a method for mapping its alternate Ids to Ensembl Ids");
+        }
+
+        return altIdToEnsemblMap;
+    }
+
+    // This uses HGNC_homologene.rpt from http://www.informatics.jax.org/downloads/reports/
+    private static Map<String, Set<String>> mapMouseAlternateIds(BufferedReader br) throws IOException {
```

Comment:
If you feel like it, you could create an common interface for doing these "alternate" mappings (which should not be too hard since all of these methods take the same input and return the same output), then create a few classes that implement it and contain only a single function for one species-specific mapping. Then, create instances of these special mappers and call their "map" function. Some might say it's cleaner than having a bunch of functions for different species all in one class, but I don't think it's necessary.